linode/manager
# Improve sorting of entity labels and regions in tables

## Description 📝

**What does this PR do?**
This PR makes two improvements to column sorting in Cloud:

- Sorting linodes, etc. by region sorts the regions by their display names rather than their API strings
- Sorting strings now uses `localeCompare()` so that, for example, `Linode 2` gets sorted before `Linode 10`

It also slightly refactors some of our sorting utils and adds unit tests.

I marked this as proof of concept in case there's some historical reason for why sorting behaves the way it does now, and in case my approach to changing this is too heavy-handed.

## Preview 📷

**Region sorting improvements:**
<img width="400" alt="region-sorting-before-after" src="https://user-images.githubusercontent.com/97627410/207964228-52a17a8b-0874-4290-b956-d4163a3bcf67.png">

**Label sorting improvements:**
<img width="640" alt="linode-label-sorting" src="https://user-images.githubusercontent.com/97627410/207964230-f33b2185-6f38-445d-9267-cf4f239bc4e1.png">

## How to test 🧪

**What are the steps to reproduce the issue or verify the changes?**
1. Navigate to Linodes landing page
2. Create some Linodes with numbered labels (e.g. `Linode 1`, `Linode 2`, `Linode 10`) across various regions
3. To confirm the issue is fixed, confirm that when sorting Linodes by label ascending, `Linode 10` gets sorted after `Linode 2`, and confirm that regions are sorted alphabetically according to their display names


**How do I run relevant unit or e2e tests?**

To run relevant unit tests:
```bash
yarn test sort-by.test
```

To run relevant e2e tests:
```bash
yarn cy:run -s "cypress/e2e/linodes/smoke-linode-landing-table.spec.ts"
```
